src/MobileRedirection.php

```diff
@@ -78,6 +86,12 @@ public function register() {
 				add_filter( 'get_comments_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For get_comments_link().
 				add_filter( 'respond_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For comments_popup_link().
 			}
+		} elseif ( AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT ) ) {
+			add_action(
+				'template_redirect',
+				[ $this, 'maybe_add_mobile_switcher_hooks' ],
+				PHP_INT_MAX
+			); // For adding the mobile switcher link when redirection is disabled.
```

Comment:
I'm just realizing that this doesn't seem to be needed. I think it is now achieved with the above condition and call to `add_action( 'wp_head', [ $this, 'add_mobile_alternative_link' ] )`. In other words, when mobile redirection is disabled _and_ loose/moderate sandboxing is enabled, _then_ we want to add the `link[rel=alternate]`. We do not also want to add the mobile switcher link, which turns out to be broken right now because `maybe_add_mobile_switcher_hooks` is not adding the required JS to do the redirection (or to show the mobile link) when on the non-AMP version.

```suggestion
```